windows-agent/internal/proservices/landscape/connection.go

```diff
@@ -218,9 +218,24 @@ func (conn *connection) receiveCommands(e executor) error {
 			return fmt.Errorf("could not receive commands: %v", err)
 		}
 
-		if err := e.exec(conn.ctx, command); err != nil {
+		// Removing the cancel context so that the command is executed even if the connection is lost.
+		ctx := context.WithoutCancel(conn.ctx)
```

Comment:
I keep forgetting about that API, this is really nice :)

Do you think we can have one test cancelling a context, and we pass that cancelled context here? (I don’t have enough "context" to see how deep we are in the stack)

Reply:
That'd be hard to exercise, line 108 right above has a cancelled context check.